### DISCUSSION TITLE
Not all metrics are returned with Prometheus

### DISCUSSION THREAD

**Discussion by singhjoga** [apache/apisix#5614] [Q&A] [answered]
Hi Guys,
I am using APISIX in Standalone mode in Kubernetes.  I have enabled Prometheus with below configuration in my config.yaml. I am able to call the /metrics endpoint successfully. However, it returns only the below given metrics. No metrics related my routes and services are returned.
<pre>
=======config==========
plugin_attr:
      prometheus:
        export_addr:
          ip: 0.0.0
          port: 9092
        prefer_name: true
        export_uri: /metrics
</pre>
<pre>
======== metrics returned ===============
# HELP apisix_nginx_http_current_connections Number of HTTP connections
# TYPE apisix_nginx_http_current_connections gauge
apisix_nginx_http_current_connections{state="accepted"} 2219
apisix_nginx_http_current_connections{state="active"} 2
apisix_nginx_http_current_connections{state="handled"} 2219
apisix_nginx_http_current_connections{state="reading"} 0
apisix_nginx_http_current_connections{state="total"} 236
apisix_nginx_http_current_connections{state="waiting"} 1
apisix_nginx_http_current_connections{state="writing"} 1
# HELP apisix_nginx_metric_errors_total Number of nginx-lua-prometheus errors
# TYPE apisix_nginx_metric_errors_total counter
apisix_nginx_metric_errors_total 0
# HELP apisix_node_info Info of APISIX node
# TYPE apisix_node_info gauge
apisix_node_info{hostname="apisix-gw-deployment-558499b786-w5ndg"} 1
================
</pre>
Any idea what I might be doing wrong?

Regards,

Joga

**Comment by singhjoga** [comment, marked as answer]
OK, after reading this article https://apisix.apache.org/blog/2021/11/04/skywalking, I was able to figure out the issue. I had not bound the plugin to the routes. It was not obvious from the plugin page https://apisix.apache.org/docs/apisix/plugins/prometheus.

**Reply by tokers** [reply]
If you'd like to collect metrics for all the requests, you can bind it to the global rule.

**Reply by juzhiyuan** [reply]
That article is written by @dmsolr, he is one of Apache SkyWalking's PMC members 😄 If you have any questions about APM, we could seek help from @dmsolr.

>  It was not obvious from the plugin page https://apisix.apache.org/docs/apisix/plugins/prometheus.

Just to make sure, do you mean this part?

```
PUT /routes
{
    "prometheus": {}
}
```
